src/components/Navbar/index.tsx

```diff
@@ -2,34 +2,69 @@
 
 import Image from 'next/image';
 import Link from 'next/link';
-import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
 
+import ROUTE from '@/constants/routes';
 import TokenHandler from '@/utils/tokenHandler';
-import { useRouter } from 'next/navigation';
+
 import logo from '../../../public/images/momentiaLogoSymbol.png';
 import Icon from '../Icon/Icon';
+import NavbarNoticeDetail from './NavbarNoticeDetail';
+import NavbarUserOption from './NavbarUserOption';
 
 const Navbar = () => {
   const router = useRouter();
   const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [isSignedIn, setIsSignedIn] = useState(
+    TokenHandler.getAccessToken() !== '',
+  );
 
-  const isSignedIn = TokenHandler.getAccessToken() !== '';
+  const moveToSignIn = () => {
+    router.push(ROUTE.signIn);
 
-  const moveToSignIn = () => router.push('/auth/sign-in');
+    if (isMenuOpen) toggleMenu();
+  };
 
   const toggleMenu = () => {
     setIsMenuOpen((prev) => !prev);
   };
 
+  const clickSignOut = () => {
+    TokenHandler.removeToken();
+    router.push(ROUTE.signIn);
+    if (isMenuOpen) toggleMenu();
+  };
+
+  useEffect(() => {
+    const handleLoginEvent = () => {
+      setIsSignedIn(TokenHandler.getAccessToken() !== '');
```

Comment:
```suggestion
      setIsSignedIn(TokenHandler.getAccessToken());
```

이것도 이렇게 해도 동일하게 작동할 듯합니다! 빈 문자열인지 체크하는 예외 처리 외에 다른 예외 처리가 없다면 불필요할 듯해여